lib/Basic/Unix/TaskQueue.inc

```diff
@@ -555,34 +562,36 @@ static bool
 cleanUpAHungUpTask(Task &T,
                    const TaskQueue::TaskFinishedCallback FinishedCallback,
                    const TaskQueue::TaskSignalledCallback SignalledCallback) {
-  const Optional<int> StatusIfOK = waitForPid(T.getPid());
-  if (!StatusIfOK)
+  const auto StatusIfOK = waitForPid(T.getPid());
```

Comment:
Please rename `StatusIfOK` to `StatusAndTimes` (assuming you change the Metrics class name to Times). Or rename to `StatusAndMetrics`.